app/models/product.rb

```diff
@@ -7,9 +7,10 @@ def validate_each(record, attribute, value)
 class Product < ApplicationRecord
   after_initialize :set_title_default
   before_save :set_discount_price_default
-  has_many :line_items
+  has_many :line_items, dependent: :restrict_with_error
   has_many :orders, through: :line_items
-  before_destroy :ensure_not_referenced_by_any_line_item
+  has_many :carts, through: :line_items
+  # before_destroy :ensure_not_referenced_by_any_line_item
```

Comment:
why commented?

Reply:
Line number 10 has the same implementation and this was the change that was asked in the question